azkaban-common/src/main/java/azkaban/server/AzkabanServer.java

```diff
@@ -75,12 +77,27 @@ public static Props loadProps(String[] args, OptionParser parser) {
       azkabanSettings = loadConfigurationFromAzkabanHome();
     }
 
+    if (azkabanSettings != null) {
```

Comment:
I agree it is. I'll fix it. We need to change the entire args processing code anyway. The current code flow is very unclean and needs to be restructured. I'll add validations when I implement the new classes. Treat this as a stop gap change.